compiler/rustc_parse/src/parser/attr_wrapper.rs

```diff
@@ -366,7 +366,8 @@ impl<'a> Parser<'a> {
             let new_tokens = vec![(FlatToken::AttrTarget(attr_data), Spacing::Alone)];
 
             assert!(
-                !self.token_cursor.break_last_token,
+                !self.token_cursor.break_last_token
+                    || (self.token.kind == TokenKind::Gt && self.unmatched_angle_bracket_count > 0),
```

Comment:
I'm not too familiar with this code but you could look into why the token isn't broken? It strikes me as unexpected that we'd give the derive a `BinOp(shr)` here instead of two `Gt` tokens?

Reply:
> why the token isn't broken?

This is intentional:

https://github.com/rust-lang/rust/blob/8882507bc7dbad0cc0548204eb8777e51ac92332/compiler/rustc_parse/src/lexer/tokentrees.rs#L59-L61